lib/jekyll/reader.rb

```diff
@@ -0,0 +1,113 @@
+# encoding: UTF-8
+require 'csv'
+
+module Jekyll
+  class Reader
+    attr_reader :site
+
+    def initialize(site)
+      @site = site
+    end
+
+    # Read Site data from disk and load it into internal data structures.
+    #
+    # Returns nothing.
+    def read
+      @site.layouts = LayoutReader.new(site).read
+      read_directories
+      @site.data = DataReader.new(site).read(site.config['data_source'])
+      CollectionReader.new(site).read
```

Comment:
I don't understand. What do you mean? Do you mean to re-read the collections every time the `@site.collections` is called?